api/glossary_usefulconcepts.md

```diff
@@ -0,0 +1,156 @@
+
+## Models
+These are an example of a Python object, specifically a database.
+
+## Relational databases
+The type of database we are interested in is called a relational database (and this is why ORM [Object Relational Mapping] is so relevant).
+Relational databases set up relationships between items efficiently, and each data are stored in a table. The rows contain the items and the columns contain the properties of the items. The ORM (in our case, SQLAlchemy) is written in a language that is part of the web framework language (in our case, Python), allowing us to communicate with the database that is more familiar than SQL.
+
+####Organisation of the data in a database versus a class:
+**Class:**
+
+|ClassName|
+|---------|
+|**id:** int|
+|**name:** str|
+|**descriptions:** str|
+
+**Relational database:**
+
+|ID:|Name:|Description:|
+|---|---|---|
+|INT|Char|VarChar|
+
+### Database keys
+Keys are fundamental in the creation of relationships between data (intra- and inter-tables). Further, they assist in maintaining uniqueness in a table; ensure consistent and valid data as well as allow for efficient data retrieval.
+
+|Candidate keys (multiple, one can be a primary key)||
+|--------------------------------------|---|
+|**Primary keys**|**Alternate key:**|
+|Uniquely identifies each record. Cannot be NULL.|Currently not selected as a primary key|
+|Ensures entity integrity (ensures there are no duplicate records within the table)|**Unique key:** |
+| |Similar to primary keys, but all for one null value|
+| |No duplicates, useful for data validation|
+| |Useful to place additional unique conditions on columns|
+|**Composite keys:**| |
+|a.k.a. Compound, composite keys| |
+| More than one columns used to identify a unique value|
+
+### Data types
+These define the type of value that can be stored in a column. 
+This summary of [PostgreSQL data types](https://www.postgresql.org/docs/9.5/datatype.html) will provide more information.
+- Numeric: 
+    - float, real, integers (no decimal, varying precision), decimal (fixed precision (p,s))
+- Characters:
+    - char, var_char, text
+- Boolean: 
+    - TRUE / FALSE
+- JSON:
+    - JavaScript Object Notation stores and exchanges data consisting of data types that are arrays or of attribute-value type. 
+- XML:
+    - Extensible markup language (creates information formats and electronically shares structured data via public and corporate networks)
+
+## Universal Resource Locators
+Protocols to locate and access information. These are used for communication between servers and clients. Examples include: [**HTTP**](https://en.wikipedia.org/wiki/Hypertext_Transfer_Protocol) (hypertext transfer protocols) and [FTP](https://en.wikipedia.org/wiki/File_Transfer_Protocol) (file transfer protocol for file transfers between client and server)
```

Comment:
Is this right? I wouldn't consider HTTP to be a URL which is what this is suggesting? Rather the communication protocol for interacting with the resource is specified part of a URL as you mentioned in the example...

Reply:
I'm really not sure. This was a point I struggled to understand - I hope the new phrasing is better